### PR TITLE
Enhance `addClass()` and `class()` methods to support `BackedEnum` for CSS classes.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Tag/Base/Tag.php [hunk 1]
```diff
@@ -81,27 +82,41 @@ final public function id(?string $id): static
...
-            array_filter($class, static fn ($c) => $c !== null),
-        );
+        foreach ($class as $c) {
+            if ($c !== null) {
+                if ($c instanceof BackedEnum) {
+                    $c = is_string($c->value) ? $c->value : null;
+                }
+                Html::addCssClass($new->attributes, $c);
+            }
+        }
```
Comment (vjik) [inline src/Tag/Base/Tag.php:97]: Add backed enumeration support to `Html::addCssClass` method instead it.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Tag/Base/Tag.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Tag/Base/Tag.php]: done.

File: src/Html.php [hunk 2]
```diff
@@ -1690,20 +1691,36 @@ public static function renderTagAttributes(array $attributes): string
      * @see removeCssClass()
      *
      * @param array $options The options to be modified.
-     * @param null[]|string|string[]|null $class The CSS class(es) to be added. Null values will be ignored.
+     * @param BackedEnum|BackedEnum[]|null[]|string|string[]|null $class The CSS class(es) to be added. Null values will
+     * be ignored.
      *
-     * @psalm-param string|array<array-key,string|null> $class
+     * @psalm-param BackedEnum|string|array<array-key,BackedEnum|string|null> $class
```
Comment (vjik) [inline src/Html.php:1697]:
```suggestion
     * @psalm-param BackedEnum|string|array<array-key,BackedEnum|string|null>|null $class
```

File: src/Html.php [hunk 3]
```diff
@@ -1690,20 +1691,36 @@ public static function renderTagAttributes(array $attributes): string
...
      */
-    public static function addCssClass(array &$options, string|array|null $class): void
+    public static function addCssClass(array &$options, BackedEnum|array|string|null $class): void
     {
         if ($class === null) {
             return;
         }
+
+        if ($class instanceof BackedEnum) {
+            $class = is_string($class->value) ? $class->value : null;
```
Comment (vjik) [inline src/Html.php:1706]:
```suggestion
            if (is_int($class->value)) {
                return;
            }
            $class = $class->value;
```

File: src/Html.php [hunk 4]
```diff
@@ -1690,20 +1691,36 @@ public static function renderTagAttributes(array $attributes): string
...
+        }
+
         if (is_array($class)) {
-            $class = array_filter($class, static fn (mixed $c): bool => $c !== null);
-            if (empty($class)) {
+            $filteredClass = [];
+            foreach ($class as $key => $value) {
+                if ($value instanceof BackedEnum) {
+                    $value = is_string($value->value) ? $value->value : null;
```
Comment (vjik) [inline src/Html.php:1713]:
```suggestion
                    $value = \is_string($value->value) ? $value->value : null;
```

File: src/Tag/Base/Tag.php [hunk 5]
```diff
@@ -81,27 +82,25 @@ final public function id(?string $id): static
     /**
      * Add one or more CSS classes to the tag.
      *
-     * @param string|null ...$class One or many CSS classes.
+     * @param BackedEnum|string|null ...$class One or many CSS classes.
      */
-    final public function addClass(?string ...$class): static
+    final public function addClass(BackedEnum|string|null ...$class): static
```
Comment (vjik) [inline src/Tag/Base/Tag.php:87]: Add this change to changelog.

File: src/Tag/Base/Tag.php [hunk 6]
```diff
@@ -81,27 +82,25 @@ final public function id(?string $id): static
...
     }
 
     /**
      * Replace current tag CSS classes with a new set of classes.
      *
-     * @param string|null ...$class One or many CSS classes.
+     * @param BackedEnum|string|null ...$class One or many CSS classes.
      */
-    final public function class(?string ...$class): static
+    final public function class(BackedEnum|string|null ...$class): static
```
Comment (vjik) [inline src/Tag/Base/Tag.php:99]: Add this change to changelog.

File: src/Html.php [hunk 7]
```diff
@@ -1690,20 +1691,36 @@ public static function renderTagAttributes(array $attributes): string
...
      * @param array $options The options to be modified.
-     * @param null[]|string|string[]|null $class The CSS class(es) to be added. Null values will be ignored.
+     * @param BackedEnum|BackedEnum[]|null[]|string|string[]|null $class The CSS class(es) to be added. Null values will
+     * be ignored.
      *
-     * @psalm-param string|array<array-key,string|null> $class
+     * @psalm-param BackedEnum|string|array<array-key,BackedEnum|string|null> $class
      */
-    public static function addCssClass(array &$options, string|array|null $class): void
+    public static function addCssClass(array &$options, BackedEnum|array|string|null $class): void
```
Comment (vjik) [inline src/Html.php:1699]: Add this change to changelog.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Tag/Base/Tag.php (hunk 5 above)
Comment (terabytesoftw) [reply src/Tag/Base/Tag.php:87]: done.

File: src/Tag/Base/Tag.php (hunk 6 above)
Comment (terabytesoftw) [reply src/Tag/Base/Tag.php:99]: done.

File: src/Html.php (hunk 7 above)
Comment (terabytesoftw) [reply src/Html.php:1699]: done.

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 ## 3.8.1 under development
 
-- no changes in this release.
+- Enh #230: Add support `BackedEnum` value to `Html::addCssClass()`, `Tag::addClass()` and `Tag::class()` methods (@terabytesoftw)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- Enh #230: Add backed enumeration value support to `Html::addCssClass()`, `Tag::addClass()` and `Tag::class()`
  methods (@terabytesoftw)
```

**Review by vjik — APPROVED** [review]

**Pull request merged**